src/components/Book.js

```diff
@@ -1,11 +1,17 @@
 import { PropTypes } from 'prop-types';
 import { useDispatch } from 'react-redux';
-import { remove } from '../redux/books/booksSlice';
+import { deleteBook } from '../api';
+import { fetchBooks, setError } from '../redux/books/booksSlice';
 
 const Book = ({ title, author, id }) => {
   const dispatch = useDispatch();
-  const clickHandler = (id) => {
-    dispatch(remove(id));
+  const clickHandler = async (id) => {
+    try {
+      await deleteBook(id);
+      dispatch(fetchBooks());
+    } catch {
+      dispatch(setError(true));
+    }
```

Comment:
**[OPTIONAL]**

- [ ] Your add and delete features are working as expected :+1: but I guess now your app is making two API calls after addition and deletion :thinking: which is not professional and not consistent with the previous milestone :eyes: If you follow the previous milestone when your app will add or delete a book then it should make one API and call and  update the data in the redux store :+1: by this approach your app will make one API call and the time complexity will be half :+1: